Use bespin-workflows API to run workflow

johnbradley opened the pull request:
Runs workflows based on data stored in bespin-workflows.
Includes large re-organization of the project.
Part of the project has been split out into https://github.com/Duke-GCB/lando-messaging.
This allows bespin-workflows to send start/cancel job requests to lando without installing all the lando requirements (openstack, cwl-runner).

The project now creates 3 scripts:

- __lando__ - server that spawns VMs and sends messages for them to run job steps
- __lando_worker__ - program that runs inside the VMs that listens for messages to run different steps
- __lando_client__ - program that can send lando the start/cancel message (only used for testing purposes)

There are now two config files:

- __/etc/lando_config.yml__ - contains all settings used by lando
- __/etc/lando_worker_config.yml__ - created and placed on the VMs by lando based off of data in /etc/lando_config.yml

dleehr commented on README.md:
````diff
@@ -4,10 +4,120 @@
+password: secret2
+queue_name: local_worker  
+```
+The queue name `local_worker` is always used for workers when `fake_cloud_service` is True in `/etc/lando_config.yml`.
````
Sounds like this explains the `fake_cloud_service` key somewhat, but some introductory text about the different services and how to run them in isolation or in concert would be helpful. 

We need some version of our whiteboard drawing in here, explaining the difference between these things. There's lando client, lando server, lando worker, and bespin API. Also lobot? Isn't that the same as lando worker?

dleehr commented on lando/server/jobapi.py:
```diff
@@ -0,0 +1,274 @@
+        Create http auth based on config passed in constructor.
+        :return: HTTPBasicAuth
+        """
+        return HTTPBasicAuth(self.settings.username, self.settings.password)
```
This is not documented well in the bespin-api project, but we have [TokenAuthentication](http://www.django-rest-framework.org/api-guide/authentication/#tokenauthentication) enabled if that's preferred over username/password: [bespin-api/bespin/settings.py](https://github.com/Duke-GCB/bespin-api/blob/eb23233f785365cb16672933011eed3fd90ac0ac/bespin/settings.py#L139).

dleehr commented:
I suppose some of the high-level overview stuff is in the PR discussion here, but it should be in code since we'll close the PR and likely not come back to it.

johnbradley commented:
I reworked the README to have more details about the big picture.
Rename job_api to bespin_api in the config file.
Changed out basic http auth out for token auth.
